File: Array/Easy/645_Set_Mismatch/Solution.py
from typing import List


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        missing = 1
        duplicate = -1

        for num in nums:
            if nums[abs(num) - 1] < 0:
                duplicate = abs(num)
            else:
                nums[abs(num) - 1] *= -1

        for i, num in enumerate(nums, 0):
            if nums[i] > 0:
                missing = i + 1
                break

        return [duplicate, missing]

File: Array/Easy/645_Set_Mismatch/Solution.cpp
#include <vector>
#include <algorithm>


class Solution {
public:
    std::vector<int> findErrorNums(std::vector<int>& nums) {
        int missing{1}, duplicate{-1};
        
        for (int num : nums) {
            if (nums[std::abs(num) - 1] < 0) duplicate = std::abs(num);
            else nums[std::abs(num) - 1] *= -1;
        }

        for (int i{}, n = nums.size(); i < n; ++i) {
            if (nums[i] > 0) {
                missing = i + 1;  
                break;
            }
        }   

        return std::vector{duplicate, missing};
    }
};